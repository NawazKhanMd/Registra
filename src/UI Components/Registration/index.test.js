import React from 'react';
import renderer from 'react-test-renderer';
import { Register } from './index';
import '@testing-library/jest-dom/extend-expect';
import { mount, shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
    location: { pathname: '/' }
  }),
}));
jest.mock('react-router', () => ({
  useHistory: () => ({
    push: jest.fn(),
    location: { pathname: '/' }
  }),
}));
jest.mock('firebase', () => ({
  initializeApp: () => ({
    analytics: jest.fn(),
    auth: jest.fn(),
  }), firestore: jest.fn(),
}));
jest.mock('react-redux', () => {
  const ActualReactRedux = require.requireActual('react-redux');
  return {
    ...ActualReactRedux,
    useSelector: jest.fn().mockImplementation(() => {
      return 1;
    }),
    useDispatch: ()=>jest.fn()
  };
});

test('Stepper Loads', () => {
  const component = renderer.create(
    <Register />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Simulating Form submit', () => {
  const wrapper = shallow((<Register />));
  const fakeEvent = { preventDefault: () => jest.fn() };
  expect(wrapper.find('form')).toHaveLength(1);
  const entries = jest.fn()
  const get = (key) => {
    switch (key) {
      case "usr_name":
      return "Md Nawaz Khan";
      case "usr_phn":
      return "9121223331";
      case "usr_email":
      return "nawazmd230@gmail.com";
      case "usr_add_1":
      return "169, Mukti";
      case "usr_add_2":
      return "Malleshwaram, Bnglr";
      case "usr_add_3":
      return "India, 560003";
      default:
        return "";
    }
  }
  const append = jest.fn()
  global.FormData = () => ({ entries, append, get })
  wrapper.find("form").simulate('submit', fakeEvent);
   expect(wrapper.find('form')).toHaveLength(1);
});


