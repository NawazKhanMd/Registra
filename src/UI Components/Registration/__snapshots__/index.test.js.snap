// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper Loads 1`] = `
<div
  className="registration_container"
>
  <form
    autoComplete="off"
    name="RegistrationForm"
    onSubmit={[Function]}
  >
    <div
      className="segment"
    >
      <h3>
        Basic Info
      </h3>
    </div>
    <div
      className="d-flex f_row justify-content-evenly f-wrap"
    >
      <label>
        <input
          name="usr_name"
          placeholder="Full Name"
          required={true}
          type="text"
        />
      </label>
      <label>
        <input
          name="usr_email"
          placeholder="Email Address"
          required={true}
          type="email"
        />
      </label>
      <label>
        <input
          maxLength="10"
          minLength="10"
          name="usr_phn"
          placeholder="Phone Number"
          required={true}
          size="10"
          type="text"
        />
      </label>
    </div>
    <div
      className="segment"
    >
      <h3>
        Address
      </h3>
    </div>
    <div
      className="d-flex f_row justify-content-evenly f-wrap"
    >
      <label>
        <input
          name="usr_add_1"
          placeholder="House Number, Street"
          required={true}
          type="text"
        />
      </label>
      <label>
        <input
          name="usr_add_2"
          placeholder="Colony, City"
          required={true}
          type="text"
        />
      </label>
      <label>
        <input
          name="usr_add_3"
          placeholder="Country, Pincode"
          required={true}
          type="text"
        />
      </label>
    </div>
    <div
      className="d-flex f_row justify-content-center"
    >
      <button
        className="red"
        type="submit"
      >
        Next
      </button>
    </div>
  </form>
</div>
`;
